### PR TITLE
[WIP] 1514: fix formatting in opcodes

### PR DESCRIPTION
**Pull request by td-bn**
See  #1514.

### CODE REVIEW

**Review by Kixunil — APPROVED** [review]
ACK dfc670a792469ba31bfd40329556f80359cea46a

Nice, thanks! Feel free to unify it with the module definition, as we discussed, if you like.

**Review by apoelstra — APPROVED** [review]
ACK e668266e38f9fc9d2fb3e4fbed7c234bfb9660c4

File: bitcoin/src/blockdata/opcodes.rs [hunk 1]
```diff
@@ -34,636 +34,299 @@ pub struct All {
     code: u8,
 }
 
-// private import so we don't have to use `all::OP_FOO` in this file.
-use self::all::*;
+macro_rules! all_opcodes {
+    ($($op:ident => $val:expr, $doc:expr);*) => {
+        // private import so we don't have to use `all::OP_FOO` in this file.
+        use self::all::*;
```
Comment (Kixunil) [inline bitcoin/src/blockdata/opcodes.rs:40]: It shouldn't matter whether this is in a macro but I don't care.

File: bitcoin/src/blockdata/opcodes.rs [hunk 2]
```diff
@@ -34,636 +34,299 @@ pub struct All {
...
+        /// Enables wildcard imports to bring into scope all opcodes and nothing else.
+        ///
+        /// The `all` module is provided so one can use a wildcard import `use bitcoin::opcodes::all::*` to
+        /// get all the `OP_FOO` opcodes without getting other types defined in `opcodes` (e.g. `All`, `Class`).
+        ///
+        /// This module is guaranteed to never contain anything except opcode constants and all opcode
+        /// constants are guaranteed to begin with OP_.
+        pub mod all {
+            use super::All;
+                $(
```
Comment (Kixunil) [inline bitcoin/src/blockdata/opcodes.rs:51]: Nit: space at the end of this line.

File: bitcoin/src/blockdata/opcodes.rs [hunk 3]
```diff
@@ -34,636 +34,299 @@ pub struct All {
...
-            OP_INVALIDOPCODE => write!(f, "INVALIDOPCODE"),
-            OP_CHECKSIGADD => write!(f, "CHECKSIGADD"),
-            All {code: x} => write!(f, "RETURN_{}", x),
-        }
-    }
+all_opcodes! {
+    OP_PUSHBYTES_0 => 0x00, "Push an empty array onto the stack.";
+    OP_PUSHBYTES_1 => 0x01, "Push the next byte as an array onto the stack.";
+    OP_PUSHBYTES_2 => 0x02, "Push the next 2 bytes as an array onto the stack.";
+    OP_PUSHBYTES_3 => 0x03, "Push the next 2 bytes as an array onto the stack.";
```
Comment (Kixunil) [inline bitcoin/src/blockdata/opcodes.rs:74]: Unrelated but Satoshi must have had some strong drugs when he decided this.

File: bitcoin/src/blockdata/opcodes.rs [hunk 4]
```diff
@@ -34,636 +34,299 @@ pub struct All {
...
+        ///
+        /// The `all` module is provided so one can use a wildcard import `use bitcoin::opcodes::all::*` to
+        /// get all the `OP_FOO` opcodes without getting other types defined in `opcodes` (e.g. `All`, `Class`).
+        ///
+        /// This module is guaranteed to never contain anything except opcode constants and all opcode
+        /// constants are guaranteed to begin with OP_.
+        pub mod all {
+            use super::All;
+                $( 
+                    /// $doc
```
Comment (Kixunil) [inline bitcoin/src/blockdata/opcodes.rs:52]: Unfortunately, this doesn't work. If I'm not mistaken `#[doc = $doc]` should work instead.

**Review by Kixunil — CHANGES_REQUESTED** [review]
Apart from broken docs, this looks fine.

**Review by Kixunil — DISMISSED** [review]
ACK ed0a055de223b763a42aaf35091b84d67f63ac75

Squashing may be a good idea though.

**Review by Kixunil — APPROVED** [review]
ACK 77799bf5cb43df5c30218c4889674aa17c798233

**Pull request merged**